spec/models/spree/product_set_spec.rb

```diff
@@ -12,7 +12,6 @@
         let(:collection_hash) do
           {
             0 => {
-              product_id: 11,
```

Comment:
Are we sure this won't ever be passed to `ProductSet` as an attribute?

Reply:
no but we can test and anyway, what does it mean? it should be :id
having a separate attribute product_id for creation is just complicating and then it's  dead code :-)

Reply:
Matt, I had a second look at both PRs. we can go with [yours](https://github.com/openfoodfoundation/openfoodnetwork/pull/4884/files) if you prefer but I think the little clean up in this one is good.

Reply:
I also much prefer the cleanup in your PR, I just have a suspicion that the test example probably represents a real case, so the :product_id attribute in the test example might still be passed to ProductSet somewhere.

Reply:
Adding details to what I have verified:
ProductSet is only used in spree/admin/products_controller on a collection scenario. I tested the bulk product update page and verified the product_id is not sent from there.
The bulk product update page does not create products in any circumstances.
The product creation happens in this controller but through ResourceController#create.